### PR TITLE
make repocard content a property

### LINKED ISSUE #1144: Make it easier to modify the model card content
Modifying `card.text` does not modify `card.content`, which is what is pushed to the Hub.

```
card = ModelCard.load('akdeniz27/roberta-large-cuad')
card.text = """
This is the new model card
"""
card.content = str(card)
```

It would work, but it has the last unintuitive step of modifying `content. Ideally, when `text` is modified, the `content` should also be modified.

cc @nateraw

### PR DESCRIPTION
**Pull request by nateraw**
Resolves #1144 by turning `RepoCard.content` into a property with an accompanying setter fn. The setter handles setting `.data` and `.text` attributes accordingly. 

This way, when you update the card text or data, the card's content will be updated as well. As an added bonus, if you update `card.content`, it will update the card data and text as well.

### CODE REVIEW

File: src/huggingface_hub/repocard.py [hunk 1]
```diff
@@ -79,7 +79,22 @@ def __init__(self, content: str):
...
+    @property
+    def content(self):
+        """The content of the RepoCard, including the YAML block and the Markdown body."""
+        line_break = _detect_line_ending(self._content) or "\n"
+        return f"---{line_break}{self.data.to_yaml(line_break=line_break)}{line_break}---{line_break}{self.text}"
+
+    @content.setter
+    def content(self, content: str):
+        """Set the content of the RepoCard."""
+        self._content = content
```
Comment (Wauplin) [inline src/huggingface_hub/repocard.py:96]: I think this private attribute is not used, right ?

**Review by Wauplin — APPROVED** [review]
Looks good to me, thanks for fixing this @nateraw !
I left a comment for a attribute that I think could be removed, otherwise PR's good for me :) 

**EDIT:** CI is broken so don't worry about the 2 failing tests. They are unrelated to this PR anyway. I'll make the CI skip them.

**Review by osanseviero — APPROVED** [review]
Nice! :fire:

File: src/huggingface_hub/repocard.py (hunk 1 above)
Comment (nateraw) [reply src/huggingface_hub/repocard.py:96]: ah yes you're right. I originally had updated to use this in `__init__` but copy pasted everything down to this setter method later. It's useless and can go. will patch real quick. 

Nice catch!! thanks

Comment (nateraw) [reply src/huggingface_hub/repocard.py:96]: EDIT - I lied. it's being used in the `content` property.

File: src/huggingface_hub/repocard.py [hunk 2]
```diff
@@ -79,7 +79,22 @@ def __init__(self, content: str):
...
         """
+
+        # Set the content of the RepoCard, as well as underlying .data and .text attributes.
+        # See the `content` property setter for more details.
         self.content = content
+
+    @property
+    def content(self):
+        """The content of the RepoCard, including the YAML block and the Markdown body."""
+        line_break = _detect_line_ending(self._content) or "\n"
```
Comment (nateraw) [inline src/huggingface_hub/repocard.py:90]: @Wauplin private attr `self._content` used here.

Comment (Wauplin) [reply src/huggingface_hub/repocard.py:90]: Ah yes, good catch !

**Pull request merged**